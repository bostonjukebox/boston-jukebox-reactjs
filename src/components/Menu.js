import React, { Component } from 'react'
import styled from 'styled-components'

import Modal from './Modal'
import ModalForm from './ModalForm'

const Container = styled.section`
    width: 100%;
    height: 100%;
`
const Navigation = styled.nav`
    position: absolute;
    bottom: 5%;
    right: 5%;
    width: 25%;
    display: flex;
    flex-wrap: nowrap;
    justify-content: space-between;
`

const Item = styled.li`
    list-style: none;
`

const Button = styled.button`
    appearance: none;
    border: none;
    background-color: transparent;
    color: #fff;
    font-size: 1rem;
    outline: none;
    cursor: pointer;

    &:hover {
        color: #000;
    }
`

class Menu extends Component {
    constructor() {
        super()
        this.state = {
            showModalWorks: false,
            showModalCredits: false,
            showModalSuggestion: false,
        }
    }

    toggleModalWorks() {
        this.setState({
            showModalWorks: !this.state.showModalWorks
        })
    }

    toggleModalCredits() {
        this.setState({
            showModalCredits: !this.state.showModalCredits
        })
    }

    toggleModalSuggestion() {
        this.setState({
            showModalSuggestion: !this.state.showModalSuggestion
        })
    }


    componentDidMount() {
        document.addEventListener('keyup', (event) => {
            if(event.keyCode === 27) {
                this.setState({showModalCredits: false})
                this.setState({showModalSuggestion: false})
                this.setState({showModalWorks: false})
            }
        })
    }

    render() {
        return(
        <Container>
            <Navigation>
                <Item><Button onClick={() => this.toggleModalSuggestion()}>suggest a song</Button></Item>
                <Item><Button onClick={() => this.toggleModalWorks()}>how it works</Button></Item>
                <Item><Button onClick={() => this.toggleModalCredits()}>credits</Button></Item>
            </Navigation>
            {(this.state.showModalCredits) ?
                <Modal
                    title={'Credits'}
                    text={'Boston TouchJukebox started thesis project of Tiago Costa under the supervising of Ravi Jain (2017). Now is a sideproject in full development ;-) \n\ncontact: tiago@tiagocosta.co'}
                    handleModal={() => this.toggleModalCredits()} />
                : ''   
            }
            {(this.state.showModalWorks) ?
                <Modal
                    title={'How it works'} 
                    text={"The Boston TouchJukebox reads your mood and what you touch to offer a unique listening experience.\n\nWhen you wear the headband sensor, the Boston TouchJukebox starts to read your mood. According to how you feel, certain songs are available. \n\nFor example, if you're stressed and touch certain objects, the Boston TouchJukebox you recommend songs to keep you more relaxed. The interface of the TouchJukebox changes according to your mood. \n\nIf you notice that colors are changing, it is a signal that you're more or less concentrated on your experience. \n\nThe volume creates the same experience. If you get concentrated, the Boston TouchJukebox gets louder to provide you a more immersive experience with the song you're listening."} 
                    handleModal={() => this.toggleModalWorks()} />
                : ''
            }
            {(this.state.showModalSuggestion) ?
                <ModalForm 
                    title={'Suggest a song'} 
                    handleModal={() => this.toggleModalSuggestion()}
                    />
                : ''
            }
        </Container>
        )
    }

}

export default Menu