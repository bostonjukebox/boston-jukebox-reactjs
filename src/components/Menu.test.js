import React from 'react'
import { shallow } from 'enzyme'
import Menu from './Menu'

test('renders without crashing', () => {
    const component = shallow(<Menu />);
    expect(component).toMatchSnapshot();
});

test('button opens modal', () => {
    const menu = shallow(<Menu />);
    menu.find(<Button></Button>).simulate('click');
    expect(menu.find(<Modal />).prop('handleModal')).toEqual(true);
})