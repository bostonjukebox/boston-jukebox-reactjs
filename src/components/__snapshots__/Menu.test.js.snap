// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.section>
    <styled.nav>
        <styled.li>
            <styled.button
                onClick={[Function]}
            >
                suggest a song
            </styled.button>
        </styled.li>
        <styled.li>
            <styled.button
                onClick={[Function]}
            >
                how it works
            </styled.button>
        </styled.li>
        <styled.li>
            <styled.button
                onClick={[Function]}
            >
                credits
            </styled.button>
        </styled.li>
    </styled.nav>
    
    
    
</styled.section>,
  "nodes": Array [
    <styled.section>
      <styled.nav>
            <styled.li>
                  <styled.button
                        onClick={[Function]}
                  >
                        suggest a song
                  </styled.button>
            </styled.li>
            <styled.li>
                  <styled.button
                        onClick={[Function]}
                  >
                        how it works
                  </styled.button>
            </styled.li>
            <styled.li>
                  <styled.button
                        onClick={[Function]}
                  >
                        credits
                  </styled.button>
            </styled.li>
      </styled.nav>
      
      
      
</styled.section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Menu />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Menu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "showModalCredits": false,
          "showModalSuggestion": false,
          "showModalWorks": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.section>
          <styled.nav>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        suggest a song
                              </styled.button>
                    </styled.li>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        how it works
                              </styled.button>
                    </styled.li>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        credits
                              </styled.button>
                    </styled.li>
          </styled.nav>
          
          
          
</styled.section>,
        "_debugID": 2,
        "_renderedOutput": <styled.section>
          <styled.nav>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        suggest a song
                              </styled.button>
                    </styled.li>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        how it works
                              </styled.button>
                    </styled.li>
                    <styled.li>
                              <styled.button
                                        onClick={[Function]}
                              >
                                        credits
                              </styled.button>
                    </styled.li>
          </styled.nav>
          
          
          
</styled.section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Menu />,
}
`;
